config/resource/config.go

```diff
@@ -0,0 +1,49 @@
+/*
+Copyright 2021 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resource
+
+import (
+	"github.com/crossplane-contrib/terrajet/pkg/config"
+
+	"github.com/crossplane-contrib/provider-tf-azure/apis/rconfig"
+)
+
+// Configure configures resource group
+func Configure(p *config.Provider) {
+	p.AddResourceConfigurator("azurerm_resource_group", func(r *config.Resource) {
+		r.Kind = "ResourceGroup"
+	})
+
+	p.AddResourceConfigurator("azurerm_resource_group_template_deployment", func(r *config.Resource) {
+		r.References = config.References{
+			"resource_group_name": config.Reference{
+				Type:      "ResourceGroup",
+				Extractor: rconfig.APISPackagePath + "/rconfig.ExtractResourceName()",
+			},
+		}
+		r.UseAsync = true
```

Comment:
```suggestion
		r.UseAsync = true
		r.Group = "resources"
		r.Kind = "GroupTemplateDeployment"
```

```
/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.Resources/deployments/template1
```

I think there are quite a few resources like this where we included them because of a coincidence with an existing group, i.e. this is an API for deploying ARM templates but we include it because it had `azurerm_resource_group` prefix. There is also `azurerm_resource_group_policy_assignment` where it actually belongs to yet another service called [`Microsoft.Authorization`](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/resource_group_policy_assignment).

So these three resources are not related at all and may not be very useful since they are just a single resource from a whole service. I think we can even remove such resources to not deal with their grouping differences.